scalafix/rules/src/main/scala/rsc/rules/RscCompat.scala

```diff
@@ -177,27 +183,44 @@ case class RscCompat(legacyIndex: SemanticdbIndex, config: RscCompatConfig)
   private def targetsForPolymorphicDefaultParams(
       env: Env,
       name: Name,
+      tparams: List[Type.Param],
       paramss: List[List[Term.Param]]
   ): Seq[RewriteDefault] = {
-    def isPolymorphic(tpe: Type): Boolean = tpe match {
+    def isPolymorphicInTypeParam(tpe: Type): Boolean = tpe match {
       case Type.Apply(_, args) =>
-        args.exists(isPolymorphic)
+        args.exists(isPolymorphicInTypeParam)
       case t =>
         t.symbol
           .flatMap(sym => symbols.info(sym.syntax))
-          .exists(_.kind == s.SymbolInformation.Kind.TYPE_PARAMETER)
+          .exists { info =>
+            tparams.exists { tpeParam =>
+              tpeParam.name.value == info.displayName
+            }
+          }
     }
 
-    val defnSymbolBase = name.symbol.get.syntax.dropRight(3) // drop "()." suffix
     for {
+      defnSymbol <- name.symbol.toSeq.map(_.syntax)
       param_i <- paramss.flatten.zipWithIndex
       (param, i) = param_i
-      default <- param.default
+      unascribedDefault <- param.default match {
+        case Some(Term.Ascribe(_, _)) => None
+        case d => d
+      }
       decltpe <- param.decltpe
-      target <- if (isPolymorphic(decltpe)) {
-        val after = default.tokens.last
-        val symbol = defnSymbolBase + "$default$" + s"${i + 1}()."
-        Some(RewriteDefault(env, after, default, symbol))
+      target <- if (isPolymorphicInTypeParam(decltpe)) {
+        val after = unascribedDefault.tokens.last
+
+        val isConstructor = defnSymbol.contains("<init>")
+        val defnSymbolBase = if (isConstructor) {
+          defnSymbol.replaceFirst("#`<init>`.*\\.", ".`<init>").reverse.dropWhile(_ != '>').reverse
+        } else {
+          defnSymbol.stripSuffix("().")
+        }
+        val maybeBacktick = if (isConstructor) "`" else ""
+        val defaultTermSymbol = defnSymbolBase + "$default$" + s"${i + 1}$maybeBacktick()."
```

Comment:
I would prefer this do be done via the symbol API, i.e. `sym.owner` and `sym.desc`, instead of string manipulation. We used to do string manipulation in the early days of Scalameta, and it may lead to very subtle errors.